example for api/simple can't compile successfully

<!--
  Please note that this template is not optional.
  Please fill out _ALL_ fields, or your issue may be closed as "invalid."
  Please do not delete this template.
  Please ask questions on StackOverflow or Gitter (https://gitter.im/webpack/webpack).
  General questions, how-to questions, and support requests will be closed.
-->

* Operating System:  Windows 10
* Node Version:  v7.10.1
* NPM Version:  4.2.0
* webpack Version:  ^4.1.1
* webpack-dev-server Version: 3.1.4

<!--
  Please place an x, no spaces, in all [ ] that apply
  Please note that we are NOT accepting FEATURE requests at this time.
-->

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

<!--
  If you have a large amount of code to share which demonstrates the problem
  you're experiencing, or your webpack config is very large, please provide a link
  to your repository rather than pasting code. We'd also encourage you to use a
  Github Gist link instead of pasting code. Otherwise, please paste relevant
  short snippets below.

  For bugs, please do consider providing a link to a stripped-down, bare-bones
  repo that can reproduce the problem you're experiencing. Many times, bugs
  aren't actual bugs, but rather specific issues with loaders, plugins, or
  an environment/OS. Problems with complicated or large applications will almost
  always require this to be triaged.
-->

```js
  // webpack.config.js
```

https://github.com/webpack/webpack-dev-server/blob/master/examples/api/simple/webpack.config.js

```js
  module.exports = setup({
  context: __dirname,
  entry: ['./app.js', '../../../client/index.js?http://localhost:8080/'],
  output: {
    filename: 'bundle.js'
  }
});
```

### Expected Behavior
 Compiled successfully and server starts up.

### Actual Behavior

hit error:

> ERROR in multi ./app.js ../../../client?http://localhost:8080/
> Module not found: Error: Can't resolve '../../../client/index.js?http://localhost:8080/' in 'C:\Users\kaiwen\webpack-dev-server-example\examples\api\simple'
>  @ multi ./app.js ../../../client?http://localhost:8080/

I can't find './../../client/index.js' at the code.
Is it deleted or renamed after an upgrade?

The server can run successfully, if remove this path:
`entry: ['./app.js'],`

Should the cli file is included?

### For Bugs; How can we reproduce the behavior?
run `node server.js` at path **examples\api\simple** 

### For Features; What is the motivation and/or use-case for the feature?

--- a/examples/util.js
+++ b/examples/util.js
@@ -11,6 +11,19 @@ const webpack = require('webpack');
 module.exports = {
   setup(config) {
     const defaults = { mode: 'development', plugins: [], devServer: {} };
+
+    if (config.entry) {
+      if (typeof config.entry === 'string') {
+        config.entry = path.resolve(config.entry);
+      } else if (Array.isArray(config.entry)) {
+        config.entry = config.entry.map((entry) => path.resolve(entry));
+      } else if (typeof config.entry === 'object') {
+        Object.entries(config.entry).forEach(([key, value]) => {
+          config.entry[key] = path.resolve(value);
+        });
+      }
+    }
+
     const result = Object.assign(defaults, config);
     const before = function before(app) {
       app.get('/.assets/*', (req, res) => {

